Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -197,4 +313,11 @@ def quantifier(vals, frall, pred):
     return frall
 
 def AllChars():
-    return [chr(i) for i in range(0x10000)]
+    return (chr(i) for i in range(0x10000))
+
+class defaults:
+    bool = staticmethod(lambda: False)
+    int = staticmethod(lambda: 0)
+    real = staticmethod(BigRational)
+    null = staticmethod(lambda: None)
```

Comment:
Consider "class" (or rather a legal munging of that), since this is a catch-all type descriptor used for any class. "null" sounds too much like a misspelling of `None` instead.